aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/GetObjectRequest.java

```diff
@@ -1008,4 +1008,47 @@ public GetObjectRequest withS3ObjectId(S3ObjectId s3ObjectId) {
         setS3ObjectId(s3ObjectId);
         return this;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) return true;
+        if (object == null || getClass() != object.getClass()) return false;
```

Comment:
Any reason you don't use `instanceof` instead of `getClass()`? It seems like it would cover the null scenario; it's also more consistent with the other equals operators in the SDK.